GH-1409: As a smith I need more design documentation publicly available

jpilgrim opened the pull request:
#1409 

I have added most of our existing design documentation here. I also added a preamble to all newly chapters that the content may be outdated. Also converted all diagrams to SVG, a few were newly created.

Since the asciispec docker image is not publicly available, I created an ant script which downloads the asciispec tool if needed. This script also uses the Chunker to split up the design document into one file per chapter, the index file contains a table of contents.

I think that we may want to add a similar script to the specification, and also use chunked versions for the webpage. But this is probably a second step.

I have run a maven build locally, since I have not set up a Jenkins here on my side yet. Got some errors, see mail to n4js-dev@eclipse.org. IMHO these failures are unrelated to my changes.

mor-n4 commented:
Just a first quick remark:
The newly added files in
```
releng/org.eclipse.n4js.releng.utils/src/org/eclipse/n4js/doctools
tests/org.eclipse.n4js.releng.utils.tests/src/org/eclipse/n4js/doctools
```
miss copyright headers.

BTW: also several of the existing files in those locations seem to miss copyright headers.

jpilgrim commented:
@mor-n4: moved more documentation as discussed, fixed failing bugs
Actually, mvn clean verify is running now on my machine with 0 failures :-)

Please review.

jpilgrim commented:
PS: If you have ant installed, just run "ant" in the design document folder :-)

mor-n4 commented on tests/org.eclipse.n4js.hlc.tests/test-catalog.json:
```diff
@@ -0,0 +1 @@
+{"sessionId":"ee5bb6a5-b7d0-4076-8125-39444c5f970a","testDescriptors":[{"origin":"DemoTest","fqn":"src-gen/BarTest/OsInspectorTest2","testMethods":["testFail"]},{"origin":"DemoTest","fqn":"src-gen/BazTest/OsInspectorTest3","testMethods":["testIgnored"]},{"origin":"DemoTest","fqn":"src-gen/FooTest/OsInspectorTest","testMethods":["testPass"]},{"origin":"DemoTest","fqn":"src-gen/subfolder/SubFolderM ...
```
I think you added this file accidentally. Maybe add to `.gitignore` to avoid this from happening in the future.

mor-n4 commented on tests/org.eclipse.n4js.transpiler.es5.tests/testdata/annotations/IDEBUG_246_DescriptionWithSingleQuote.n4js.xt:
```diff
@@ -26,13 +26,17 @@ stderr:
 
+// Remark: avoid using x.toString() calls since different node versions emit different string literals!
+
 console.log(A)
```
But this is still using `x.toString()`, isn't it?
Does not seem to cause issues (between the node versions we and you are using, atm) but still having this line in place makes the newly added comment a bit confusing.

(same in the other files starting with `IDEBUG_246_*` in this folder)

jpilgrim replied:
To be more accurate: toString() on Objects is a problem, not on Strings

mor-n4 replied:
But `A` in line `console.log(A);` evaluates to a class constructor, not a string. Or am I overlooking something?

mor-n4 commented on docs/org.eclipse.n4js.design/chapters/02_eclipseSetup/eclipseSetup.adoc:
```diff
@@ -24,24 +28,47 @@ This installer is to be downloaded from https://wiki.eclipse.org/Eclipse_Install
-- Xpect 0.1
+- Xtext/Xtend 2.17.0
+- Xcore 1.9.0
+- Xpect 0.2
```
It's Xtext 2.18.0 and Xcore 1.10.0, at the moment.
For Xpect, people would have to use version `0.2.0.201906240918` from
```
https://ci.eclipse.org/xpect/job/Xpect-Integration-Release/20/artifact/org.eclipse.xpect.releng/p2-repository/target/repository/
```
